lib/ddtrace/workers/trace_writer.rb

```diff
@@ -194,6 +196,22 @@ def write(trace)
         # Queue the trace if running asynchronously, otherwise short-circuit and write it directly.
         async? ? enqueue(trace) : write_traces([trace])
       end
+
+      private
+
+      def record_metrics
+        if Utils::Time::THREAD_CPU_TIME_SUPPORTED
+          Datadog.health_metrics.writer_cpu_time { thread_cpu_time_diff }
+        end
+      end
+
+      # Difference in seconds of CPU time since last measurement
+      def thread_cpu_time_diff
```

Comment:
Does this need to be here and the `AsyncTransport`?

Reply:
👋  @brettlangdon, are you asking why we have it twice?
If so, it's because we have the current worker thread but also a new worker thread rewrite already merged (but not active).